### PR TITLE
feat(v0): add presence

### PR DESCRIPTION
**Pull request by Steellgold**
## Description 
Added the presence

## Acknowledgements
- [X] I read the [Presence Guidelines](https://github.com/PreMiD/Presences/blob/main/.github/CONTRIBUTING.md)
- [X] I linted the code by running `yarn format`
- [X] The PR title follows the repo's [commit conventions](https://github.com/PreMiD/Presences/blob/main/.github/COMMIT_CONVENTION.md)

## Screenshots
<details>
<summary> Proof showing the creation/modification is working as expected </summary>

image
image
image

</details>

### CODE REVIEW

**Comment by Steellgold** [comment]
image

https://github.com/PreMiD/Presences/actions/runs/6460632640/job/17538716799?pr=7654

🥸

**Comment by Bas950** [comment]
> image
> 
> https://github.com/PreMiD/Presences/actions/runs/6460632640/job/17538716799?pr=7654
> 
> 🥸

Schema version is not up to date - expected: https://schemas.premid.app/metadata/1.9, got: https://schemas.premid.app/metadata/1.7

**Comment by Steellgold** [comment]
> > image
> > https://github.com/PreMiD/Presences/actions/runs/6460632640/job/17538716799?pr=7654
> > 🥸
> 
> Schema version is not up to date - expected: https://schemas.premid.app/metadata/1.9, got: https://schemas.premid.app/metadata/1.7

Concretely, what do I have to modify? I don't really understand.

**Comment by Steellgold** [comment]
@Bas950 Fixed in https://github.com/PreMiD/Presences/pull/7654/commits/97621c16dc6bbdd9dfa24a47dd9ff141e4aefdc7

But: "HTTPError: Response code 403 (rate limit exceeded)" in the github action

File: websites/V/v0/presence.ts [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+			? document.location.pathname.slice(
+					0,
+					document.location.pathname.length - 1
+			  )
+			: document.location.pathname
+	) {
+		case "/":
+			presenceData.details = "Viewing homepage";
+			presenceData.largeImageKey = "https://i.imgur.com/UhCCRnQ.png";
+			presenceData.smallImageKey = "";
```
Comment (Bas950) [inline websites/V/v0/presence.ts:20]:
```suggestion
			delete presenceData.smallImageKey;
```

File: websites/V/v0/presence.ts [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+			break;
+		case `/r/${document.location.pathname.split("/")[2]}`:
+			presenceData.details = "Prompting a template";
+			presenceData.state = `User prompt: "${document
+				.querySelector("span.text-sm.line-clamp-1.text-ellipsis.text-left")
+				.textContent.slice(0, 50)}"`;
+			presenceData.buttons = null;
```
Comment (Bas950) [inline websites/V/v0/presence.ts:27]:
```suggestion
			delete presenceData.buttons;
```

File: websites/V/v0/presence.ts [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+				},
+			];
+			break;
+		// v0.dev/{username}
+		case `/${document.location.pathname.split("/")[1]}`:
+			presenceData.details = `Viewing @${
+				document.location.pathname.split("/")[1]
+			} profile`;
+			presenceData.largeImageKey = (
+				document.querySelector("img.rounded-full.h-12.w-12") as HTMLImageElement
```
Comment (Bas950) [inline websites/V/v0/presence.ts:53]:
```suggestion
				document.querySelector<HTMLImageElement>("img.rounded-full.h-12.w-12")
```

File: websites/V/v0/presence.ts [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+		case `/r/${document.location.pathname.split("/")[2]}`:
+			presenceData.details = "Prompting a template";
+			presenceData.state = `User prompt: "${document
+				.querySelector("span.text-sm.line-clamp-1.text-ellipsis.text-left")
+				.textContent.slice(0, 50)}"`;
+			presenceData.buttons = null;
+			break;
+		case `/t/${document.location.pathname.split("/")[2]}`:
+			presenceData.details = `Viewing a template by @${
+				(document.querySelector("a.flex-none") as HTMLAnchorElement).href.split(
```
Comment (Bas950) [inline websites/V/v0/presence.ts:31]:
```suggestion
				document.querySelector<HTMLAnchorElement>("a.flex-none").href.split(
```

File: websites/V/v0/presence.ts [hunk 5]
```diff
@@ -0,0 +1,65 @@
+const presence = new Presence({ clientId: "1160993221854380132" }),
```
Comment (Bas950) [inline websites/V/v0/presence.ts:1]:
```suggestion
const enum Assets {
	Logo = "https://i.imgur.com/UhCCRnQ.png",
}

const presence = new Presence({ clientId: "1160993221854380132" }),
```

File: websites/V/v0/presence.ts [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+	presenceData: PresenceData = {
+		largeImageKey: "https://i.imgur.com/UhCCRnQ.png",
```
Comment (Bas950) [inline websites/V/v0/presence.ts:3]:
```suggestion
		largeImageKey: Assets.Logo,
```

File: websites/V/v0/presence.ts [hunk 7]
```diff
@@ -0,0 +1,65 @@
...
+		document.location.pathname.length > 1
+			? document.location.pathname.slice(
+					0,
+					document.location.pathname.length - 1
+			  )
+			: document.location.pathname
+	) {
+		case "/":
+			presenceData.details = "Viewing homepage";
+			presenceData.largeImageKey = "https://i.imgur.com/UhCCRnQ.png";
```
Comment (Bas950) [inline websites/V/v0/presence.ts:19]:
```suggestion
			presenceData.largeImageKey = Assets.Logo;
```

File: websites/V/v0/presence.ts [hunk 8]
```diff
@@ -0,0 +1,65 @@
...
+			break;
+		// v0.dev/{username}
+		case `/${document.location.pathname.split("/")[1]}`:
+			presenceData.details = `Viewing @${
+				document.location.pathname.split("/")[1]
+			} profile`;
+			presenceData.largeImageKey = (
+				document.querySelector("img.rounded-full.h-12.w-12") as HTMLImageElement
+			).src;
+			presenceData.smallImageKey = "https://i.imgur.com/UhCCRnQ.png";
```
Comment (Bas950) [inline websites/V/v0/presence.ts:55]:
```suggestion
			presenceData.smallImageKey = Assets.Logo;
```

**Review by Bas950 — CHANGES_REQUESTED** [review]
Please do test these changes locally, but some small improvements, for the rest lgtm

**Comment by Bas950** [comment]
> @Bas950 Fixed in [97621c1](https://github.com/PreMiD/Presences/commit/97621c16dc6bbdd9dfa24a47dd9ff141e4aefdc7)
> 
> But: "HTTPError: Response code 403 (rate limit exceeded)" in the github action

Yea sorry that happens sometimes.

File: websites/V/v0/presence.ts (hunk 8 above)
Comment (Steellgold) [reply websites/V/v0/presence.ts:55]: ".Logo" not exist

**Comment by Steellgold** [comment]
image

**Comment by Steellgold** [comment]
> Please do test these changes locally, but some small improvements, for the rest lgtm

Work

**Review by Bas950 — APPROVED** [review]

**Review by theusaf — APPROVED** [review]

**Pull request merged**